ai_transform/api/wrappers.py

```diff
@@ -35,28 +38,41 @@ def request_wrapper(
     if retry_func is None:
         retry_func = lambda result: False
 
+    result: requests.Response
+
     for n in range(1, num_retries + 1):
         try:
             result = fn(*args, **kwargs)
 
-            if result.status_code != 200:
+            if not result.ok:
                 to_log = format_logging_info({"message": result.content.decode(), "status_code": result.status_code})
                 if output_to_stdout:
                     print(to_log)
-                else:
-                    logger.debug(to_log)
-
                 raise ValueError(to_log)
 
             if retry_func(result):
                 to_log_for_retry = "Manual Retry Triggered..."
                 if output_to_stdout:
                     print(to_log_for_retry)
-                else:
-                    logger.debug(to_log_for_retry)
-                raise ManualRetry
-
-        except (Exception, ManualRetry) as e:
+                raise ManualRetry(to_log_for_retry)
+
+            if is_json_decodable or key_for_error:
```

Comment:
I think this can be placed into a separate function so that this request wrapper doesn't end up being 100 lines of code. It would make for easier debugging. Ideally functions should be < 10 lines of code each where possible.